### PR TITLE
Reorganizing fix library

### PR DESCRIPTION
**Pull request by winterheart**
Fixed point math library was part of the Vector math library until it was splitted to its own. Before that there was bundled version of vecmat code in scripts. This PR makes the library an isolated submodule that can be linked to any other.

Additional enhancements includes fixing minor warnings and removing unused Mac OS and Windows asm code.

### CODE REVIEW

**Review by Lgt2x — APPROVED** [review]
Looks legit. Tested on linux. Maybe a second pair of eyes could be useful @jcoby   @Arcnor ?

File: fix/fix.cpp [hunk 1]
```diff
@@ -100,9 +97,6 @@ void InitMathTables() {
 
   asin_table[256] = asin_table[255];
   acos_table[256] = acos_table[255];
-
-  //	Initialize a random seed.
-  ps_srand(time(NULL));
```
Comment (Arcnor) [inline fix/fix.cpp:105]: So this is not needed to initialize the random number generator?

Comment (Lgt2x) [reply fix/fix.cpp:105]: they relocated seed initialization to init.cpp

Comment (winterheart) [reply fix/fix.cpp:105]: Random seed was injected into InitMathTables when it was part of vecmat submodule. Logically seed initialization not belongs to vecmat or fixmat components so I decided to move it out to common initialization function.

**Pull request merged**